client/src/main/java/com/aws/greengrass/cli/commands/TopicCommand.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.cli.commands;
+
+import com.aws.greengrass.cli.adapter.NucleusAdapterIpc;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import picocli.CommandLine;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import static com.aws.greengrass.cli.adapter.impl.NucleusAdapterIpcClientImpl.deTilde;
+
+@CommandLine.Command(name = "topic", resourceBundle = "com.aws.greengrass.cli.CLI_messages",
+        subcommands = CommandLine.HelpCommand.class, mixinStandardHelpOptions = true,
+        versionProvider = com.aws.greengrass.cli.module.VersionProvider.class)
+public class TopicCommand extends BaseCommand {
+    private final NucleusAdapterIpc nucleusAdapterIpc;
+
+    private static final String LOCAL = "LOCAL";
+    private static final String MQTT = "MQTT";
+
+    private enum MSGTYPE{
+        LOCAL,
+        MQTT
+    }
+
+    public static class MessageType{
+        @CommandLine.Option(names = {"--pubsub"}, paramLabel = "To send local message.",
+                required = true, arity = "0..1", fallbackValue = LOCAL) MSGTYPE local;
+
+        @CommandLine.Option(names = {"--iotcore"}, paramLabel = "To send Iot Core message.",
+                required = true, arity = "0..1", fallbackValue = MQTT) MSGTYPE mqtt;
+    }
+
+    @Inject
+    public TopicCommand(NucleusAdapterIpc nucleusAdapterIpc) {
+        this.nucleusAdapterIpc = nucleusAdapterIpc;
+    }
+
+    @CommandLine.Command(name = "pub",
+            description = "Publish message to the specific topic ",
+            mixinStandardHelpOptions = true,
+            versionProvider = com.aws.greengrass.cli.module.VersionProvider.class)
+    public void pub(@CommandLine.Option(names = {"-t", "--topic"}, paramLabel = "The name of the topic.", required = true) String topicName,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 89-96 into a separate method.